Code/GraphMol/AddHs.cpp

```diff
@@ -707,6 +707,9 @@ void removeHs(RWMol &mol, const RemoveHsParameters &ps, bool sanitize) {
     if (!ps.removeMapped && atom->getAtomMapNum()) {
       continue;
     }
+    if (!ps.removeHydrides && atom->getAtomicNum() == 1 && atom->getFormalCharge() == -1) {
```

Comment:
Ah, missed this one. Sorry:
you don't need to test the atomic number here. that's taken care of above on line 683.